apps/sms/js/thread_list_ui.js

```diff
@@ -408,89 +477,90 @@ var ThreadListUI = {
   // please make sure url will also be revoked if new delete api remove threads
   // without calling removeThread in the future.
   delete: function thlui_delete() {
-    function performDeletion() {
-      /* jshint validthis: true */
+    var selected = this.selectionHandler.selectedList;
 
-      var threadIdsToDelete = [],
-          messageIdsToDelete = [],
-          threadCountToDelete = 0,
-          selected = this.selectionHandler.selectedList;
-
-      function exitEditMode() {
-        ThreadListUI.cancelEdit();
-        WaitingScreen.hide();
+    return Utils.confirm(
+      {
+        id: 'deleteThreads-confirmation-message',
+        args: { n: this.selectionHandler.selectedCount }
+      },
+      null,
+      {
+        text: 'delete',
+        className: 'danger'
       }
+    ).then(ThreadListUI.performDeletion.bind(this, selected));
```

Comment:
I think it's more readable to do: `.then(() => this.performDeletion(selected))`

The bonus is that if performDeletion returns a promise, then the promise returned by delete will have the same status and will wait for performDeletion result too.